File: storage_integration/storage_integration/page/storage_backups/storage_backups.js
frappe.pages['storage-backups'].on_page_load = function(wrapper) {
	var page = frappe.ui.make_app_page({
		parent: wrapper,
		title: 'Storage Integration Backups',
		single_column: true
	});

	page.add_inner_button(__("Create Backup"), function () {
		frappe.call({
			method: "storage_integration.storage_integration.page.storage_backups.storage_backups.create_backups",
			callback: (r) => {
				frappe.msgprint("Backup scheduled for now. You can download it from this page once it is complete.")
			},
		});
	});

	$(frappe.render_template("storage_backups")).appendTo(page.body.addClass("no-border"));
}

File: storage_integration/storage_integration/page/storage_backups/storage_backups.py
import frappe
from storage_integration.controller import MinioConnection


def get_context(context):
	# get backup urls from s3
	# create a download_backup function and download backup from
	files = frappe.get_all("Storage Backup", ["file_name", "key", "date"])

	return {"files": files}


@frappe.whitelist()
def create_backups():
	from frappe.utils.backups import backup

	backup_files = backup(with_files=False)
	path = backup_files["backup_path_db"]

	minio = MinioConnection(None)
	minio.upload_backup(path)
